sdk/storage/azure-storage-blob/tests/test_blob_tags.py

```diff
@@ -103,6 +103,29 @@ def test_set_blob_tags(self, resource_group, location, storage_account, storage_
         # Assert
         self.assertIsNotNone(resp)
 
+    @pytest.mark.playback_test_only
+    @GlobalStorageAccountPreparer()
+    def test_set_blob_tags_with_lease(self, resource_group, location, storage_account, storage_account_key):
+        self._setup(storage_account, storage_account_key)
+        blob_client, _ = self._create_block_blob()
+        lease = blob_client.acquire_lease()
+
+        # Act
+        blob_tags = {"tag1": "firsttag", "tag2": "secondtag", "tag3": "thirdtag"}
+        with self.assertRaises(HttpResponseError):
+            blob_client.set_blob_tags(blob_tags)
+        blob_client.set_blob_tags(blob_tags, lease=lease)
+
+        blob_client.get_blob_tags()
```

Comment:
Is this supposed to work? the blob client is leased yet it doesn't need it? I assume read operations don't need the lease? (fails if a wrong lease is given instead)

Reply:
Resolved offline.